### PR TITLE
Add eventgrid sample

### PR DESCRIPTION
**Pull request by Netyyyy**
## Purpose

As title,
add sample for `spring-cloud-azure-starter-eventgrid`.

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+This sample demonstrates how to use EventGrid to send EventGridEvent to a topic and use Service Bus Queue as Event Handlers to receive.
+
+## What You Will Build
+You will build an application with EventGrid to send EventGridEvent to a topic and use Service Bus Queue as Event Handlers to receive.
+
+## What You Need
+
+- [An Azure subscription](https://azure.microsoft.com/free/)
+- [JDK8](https://www.oracle.com/java/technologies/downloads/) or later
+- Maven
```
Comment (saragluna) [inline eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/README.md:22]: Can we use maven wrapper instead? Or does this repo all use Maven directly?

File: eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/src/main/java/com/azure/spring/sample/eventgrid/EventGridController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @GetMapping("/publish")
+    public String publish() {
+
+        // Create a EventGridEvent with String data
+        String str = "FirstName: John, LastName: James";
+        EventGridEvent event = new EventGridEvent("A user is created", "User.Created.Text", BinaryData.fromObject(str), "0.1");
+
+        client.sendEvent(event);
+
+        return "Publish success.";
```
Comment (saragluna) [inline eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/src/main/java/com/azure/spring/sample/eventgrid/EventGridController.java:27]:
```suggestion
        return "Publishing succeeds.";
```

File: eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/src/main/java/com/azure/spring/sample/eventgrid/EventGridSampleApplication.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(EventGridSampleApplication.class);
+    public static void main(String[] args) {
+        SpringApplication.run(EventGridSampleApplication.class, args);
+    }
+
+    @Bean
+    public Consumer<Message<String>> consume() {
+        return message -> {
+            List<EventGridEvent> eventDatas = EventGridEvent.fromString(message.getPayload());
```
Comment (saragluna) [inline eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/src/main/java/com/azure/spring/sample/eventgrid/EventGridSampleApplication.java:27]:
```suggestion
            List<EventGridEvent> eventData = EventGridEvent.fromString(message.getPayload());
```

data is the plural

**Review by saragluna — CHANGES_REQUESTED** [review]

File: eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/README.md (hunk 1 above)
Comment (Netyyyy) [reply eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/README.md:22]: This repo all use Maven, shall we replace them all?

Comment (saragluna) [reply eventgrid/spring-cloud-azure-starter-eventgrid/spring-cloud-azure-sample-eventgrid/README.md:22]: Not now.

**Review by saragluna — APPROVED** [review]

**Pull request merged**